docs/deep-dive/scaffolding/your-first-scaffold-plugin.md

```diff
@@ -0,0 +1,138 @@
+---
+id: your-first-scaffold-plugin
+title: Your first scaffold plugin
+sidebar_label: Creating scaffold plugins
+---
+
+You've already seen our scaffold templates, but you desire more. You want to create your own packages that improve your project.
+
+We have good news: this is completely possible and easy to accomplish using our [Plugins system](/docs/deep-dive/plugins-crash-course).
+
+# Alright, how do I do it?
+First of all, you have to update the `/webiny.root.js` file found in your project's root with your new plugin. Plugins referenced here are mainly resolved using [node module resolution](https://nodejs.org/api/modules.html). 
+
+In this example, we'll create a **local package** called **scaffoldFolder**.
+
+You may also use **globally installed packages** and, if you're willing to go as far as to write gross, indecent yet quick **inline** hacky plugins, we've made sure that's possible too.
+
+Let's see how **webiny.root.js** changes after adding our plugin:
+
+- Our original file
+```js
+module.exports = {
+    projectName: "webiny-js",
+    cli: {
+        plugins: [
+            "@webiny/cli-plugin-deploy-components",
+            "@webiny/cli-plugin-scaffold",
+            "@webiny/cli-plugin-scaffold-graphql-service",
+            "@webiny/cli-plugin-scaffold-lambda"
+        ]
+    }
+};
+```
+- **Local / Global package** (which exports one or multiple plugins)
+```js
+module.exports = {
+    projectName: "webiny-js",
+    cli: {
+        plugins: [
+            "@webiny/cli-plugin-deploy-components",
+            "@webiny/cli-plugin-scaffold",
+            "@webiny/cli-plugin-scaffold-graphql-service",
+            "@webiny/cli-plugin-scaffold-lambda",
+            "./scaffoldFolder",
+        ]
+    }
+};
+```
+- **Inline plugin**
+```js
+module.exports = {
+    projectName: "webiny-js",
+    cli: {
+        plugins: [
+            "@webiny/cli-plugin-deploy-components",
+            "@webiny/cli-plugin-scaffold",
+            "@webiny/cli-plugin-scaffold-graphql-service",
+            "@webiny/cli-plugin-scaffold-lambda",
+            {
+                name: "cli-plugin-scaffold-template-scaffold-folder",
+                type: "cli-plugin-scaffold-template",
+                scaffold: {
+                    name: "Scaffold folder",
+                    questions: () => [
+                        /* ... */
+                    ],
+                    generate: () => {
+                        /* ... */
+                    }
+                }
+            }
+        ]
+    }
+};
+```
+
+# Scaffold plugin structure
+As you may see in the inline example above, the plugin's **type** must always be **cli-plugin-scaffold-template**.
+But more signifitcantly, you'll provide a **scaffold** object that contains three fields:
```

Comment:
IMHO this copy is more clear, what do you say @webiny/webinydevteam ?

Other than `name` and `type`, you'll need to provide a **scaffold** object that contains three fields:

Reply:
done